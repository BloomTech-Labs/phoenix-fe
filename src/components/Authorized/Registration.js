import React, { useState } from 'react';
import { withRouter } from 'react-router-dom';
import Modal from '@material-ui/core/Modal';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { axiosWithAuth } from '../../utils/axiosWithAuth.js';
import { ModalStyle, getModalStyle } from '../../styles/ModalFormStyles.js';

const useStyles = ModalStyle;

const Registration = props => {
  const [credentials, setCredentials] = useState({
    username: '',
    password: '',
    name: '',
    email: '',
    age: ''
  });

  const handleChange = event => {
    setCredentials({
      ...credentials,
      [event.target.name]: event.target.value
    });
  };

  const handleSubmit = async event => {
    event.preventDefault();
    handleClose();
    axiosWithAuth()
      .post('/auth/register', credentials)
      .then(res => {
        console.log(res);
        props.history.push('/dashboard');
      })
      .catch(err => console.log('ERROR MESSAGE', err.response));
  };

  const classes = useStyles();
  const [modalStyle] = useState(getModalStyle);
  const [open, setOpen] = useState(false);

  const handleOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <>
      <Button variant="outlined" onClick={handleOpen}>
        {props.title}
      </Button>

      <Modal
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
        open={open}
        onClose={handleClose}
      >
        <div style={modalStyle} className={classes.paper}>
          <h2 id="simple-modal-title">Register</h2>
          <p id="simple-modal-description">
            Birds of a feather... Join to find Phoenix meets near you
          </p>
          <div>
            <form className={classes.root} onSubmit={handleSubmit}>
              <TextField
                required
                id="filled-required1"
                name="username"
                label="Username"
                value={props.username}
                onChange={handleChange}
                placeholder="Username"
                variant="outlined"
              />
              <TextField
                required
                id="filled-required2"
                name="password"
                label="Password"
                type="password"
                minLength="8"
                value={props.password}
                onChange={handleChange}
                placeholder="Must be 8 characters"
                variant="outlined"
              />
              <TextField
                required
                id="filled-required3"
                name="name"
                label="Full Name"
                value={props.name}
                onChange={handleChange}
                placeholder="Letters are nice"
                variant="outlined"
              />
              <TextField
                required
                id="filled-required4"
                name="email"
                label="Email"
                type="email"
                value={props.email}
                onChange={handleChange}
                placeholder="@ required"
                variant="outlined"
              />
              <TextField
                required
                id="filled-required5"
                name="age"
                type="number"
                label="Age"
                value={props.age}
                onChange={handleChange}
                variant="outlined"
              />
              <button onSubmit={handleSubmit}>Register</button>
            </form>
          </div>
        </div>
      </Modal>
    </>
  );
};

export default withRouter(Registration);
