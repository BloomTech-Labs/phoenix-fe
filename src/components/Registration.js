import React, { useState } from 'react';
import  Modal  from '@material-ui/core/Modal';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import axios from 'axios';
// import Avatar from '@material-ui/core/Avatar';

import { makeStyles } from '@material-ui/core/styles';



function rand() {
    return Math.round(Math.random() * 20) - 10;
  }
  
  function getModalStyle() {
    const top = 50 + rand();
    const left = 50 + rand();
  
    return {
      top: `${top}%`,
      left: `${left}%`,
      transform: `translate(-${top}%, -${left}%)`,
    };
  }
  
  const useStyles = makeStyles(theme => ({
    paper: {
      position: 'absolute',
      width: 400,
      backgroundColor: theme.palette.background.paper,
      border: '2px solid #000',
      boxShadow: theme.shadows[5],
      padding: theme.spacing(2, 4, 3),
    },
    root: {
      '& .MuiTextField-root': {
        display:"flex", 
        margin: theme.spacing(1),
        width: 200
        
      },
    }
  }));



function Registration (props) {
  const [ credentials, setCredentials ] = useState({
    user: { 
        username: '',
        password: '',
        name:'',
        email:'',
        age: ''
    }
})
const handleChange = (event) => {
  setCredentials({
      ...credentials, [event.target.name]: event.target.value
  });
}
    const classes = useStyles();
    const [modalStyle] = useState(getModalStyle);
    const [open, setOpen] = useState(false);
    
    const handleOpen = () => {
        setOpen(true);
      };
    
      const handleClose = () => {
        setOpen(false);
      };
      const handleSubmit = e => {
        e.preventDefault();
        axios
          .post("http://localhost:5555/api/auth/register", {
            username: credentials.username,
            password: credentials.password,
            name:credentials.name,
            email:credentials.email,
            age: credentials.age

          })
          .then(function(response) {
            console.log(response);
            localStorage.setItem("credentials", response);
            props.history.push("/login");
          })
          .catch(function(error) {
            console.log(error);
          });
      };
      return (
          <>
        <div>
        <Button variant="outlined" onClick={handleOpen}>Register</Button>
        
          <Modal
            aria-labelledby="simple-modal-title"
            aria-describedby="simple-modal-description"
            open={open}
            onClose={handleClose}
          >
            <div style={modalStyle} className={classes.paper}>
              <h2 id="simple-modal-title">New Phoenix </h2>
              <p id="simple-modal-description">
              Start Your Journey As A Phoenix 
              </p>
              <div>
                <form className={classes.root} onSubmit={handleSubmit}>
              <div>
              {/* <Avatar src="/broken-image.jpg" /> */}
              <TextField
                  required
                  id="filled-required"
                  label="Username"
                  value={props.username} 
                  onChange={handleChange}
                  placeholder="Username"
                  variant="outlined" />
                  
                  </div>
                   <TextField
                  required
                  id="filled-required"
                  label="Password"
                  value={props.password} 
                  onChange={handleChange}
                  placeholder="Must be 8 characters"
                  variant="outlined"
                  />
                    <TextField
                  required
                  id="filled-required"
                  label="Full Name"
                  value={props.name} 
                  onChange={handleChange}
                  placeholder='Letters are nice'
                  variant="outlined"
                  />
                    <TextField
                  required
                  id="filled-required"
                  label="Email"
                  value={props.email} 
                  onChange={handleChange}
                  placeholder="@ required"
                  variant="outlined"
                  /> 
                   <TextField
                  required
                  id="filled-required"
                  label="Age"
                  value={props.age} 
                  onChange={handleChange}
                  variant="outlined"
                  />
                  
                  </form>
                  <Button variant="outlined">Register</Button>
              </div>
            </div>
          </Modal>
        </div>
        </>
      );
      }
    

export default Registration;