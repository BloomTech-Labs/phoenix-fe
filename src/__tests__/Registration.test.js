import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'
import Registration from '../components/Registration.js';

it("Renders Registraion page to match snapshot", () => {
  const tree = renderer

    .create(<Router><Registration /></Router>)
    .toJSON();

    console.log(tree, 'tree')

  expect(tree).toMatchSnapshot()
});
