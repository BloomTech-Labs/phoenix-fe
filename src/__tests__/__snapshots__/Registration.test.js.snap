// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Registraion page to match snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
</button>
`;
